txnkv/txnlock/lock_resolver.go

```diff
@@ -220,9 +220,55 @@ func (lr *LockResolver) getResolved(txnID uint64) (TxnStatus, bool) {
 	return s, ok
 }
 
+// BatchResolveLegacyLocks tries to resolve legacy locks with batch method.
+// Travesal the given locks and check that:
+// 1. If the tts of lock is equal with or smaller than safepointTS, it will
+// rollback the txn, no matter the lock is expired of not.
+// 2. If the tts of lock is larger than safepointTS, it will check status of the txn.
+// Resolve the lock if txn is expired, Or do nothing.
+func (lr *LockResolver) BatchResolveLegacyLocks(
+	bo *retry.Backoffer,
+	locks []*Lock,
+	loc locate.RegionVerID,
+	safepoint uint64,
+	lowResolutionTS uint64,
+) (bool, error) {
+	if len(locks) == 0 {
+		return true, nil
+	}
+
+	locksBeforeSafePoint := make([]*Lock, 0, len(locks))
+	locksAfterSafePoint := make([]*Lock, 0, len(locks))
+	for _, l := range locks {
+		if l.TxnID <= safepoint {
+			locksBeforeSafePoint = append(locksBeforeSafePoint, l)
+		} else {
+			locksAfterSafePoint = append(locksAfterSafePoint, l)
+		}
+	}
+
+	logutil.BgLogger().Info("BatchResolveLegacyLocks",
+		zap.Uint64("safepoint", safepoint),
+		zap.Uint64("low-resolution-ts", lowResolutionTS),
+		zap.Int("before-safepoint-count", len(locksBeforeSafePoint)),
+		zap.Int("after-safepoint-count", len(locksAfterSafePoint)))
+
+	// Use math.MaxUint64 as usedCheckTTL, which means rollback the txn, no matter the lock is expired or not!
```

Comment:
Is `usedCheckTTL` an old name that you forgot to change?

Reply:
This field `usedCheckTTL` is used for check whether is the lock with TTL expired.
Maybe a better name, what about you ?